src/webapp/reports/autogenerated-forms/widgets/TextWidget.tsx

```diff
@@ -13,27 +13,55 @@ const TextWidget: React.FC<TextWidgetProps> = props => {
     const { onValueChange, dataValue, disabled } = props;
 
     const [stateValue, setStateValue] = React.useState(dataValue.value);
+    const [emailError, setEmailError] = React.useState<string | undefined>();
     React.useEffect(() => setStateValue(dataValue.value), [dataValue.value]);
 
-    const updateState = React.useCallback(({ value }: { value: string }) => {
-        setStateValue(value);
+    const validateEmail = React.useCallback((email: string) => {
```

Comment:
Can we move this function outside the component? Right now there's no need to check for changes in every re-render.